pkgs/applications/misc/j4-dmenu-desktop/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, cmake, dmenu }:
+
+stdenv.mkDerivation rec {
+  name    = "j4-dmenu-desktop-${version}";
+  version = "2.14";
+
+  src = fetchFromGitHub {
+    owner  = "enkore";
+    repo   = "j4-dmenu-desktop";
+    rev    = "r${version}";
+    sha256 = "14srrkz4qx8qplgrrjv38ri4pnkxaxaq6jy89j13xhna492bq128";
+  };
+
+  patchPhase = ''
+    substituteInPlace ./src/Main.hh \
+      --replace "\"dmenu -i\"" "\"${dmenu}/bin/dmenu -i\""
+  '';
+
+  buildInputs = [ cmake dmenu ];
```

Comment:
Changing to
```nix
postPatch = ''
  sed -e 's,dmenu -i,${dmenu}/bin/dmenu -i,g' -i ./src/Main.hh
''
```

yields
> $nix-store -qR ./result
/nix/store/zsk34dk2wdzkryllmhfyh1bqkp77lsn6-glibc-2.23
/nix/store/fls104kb2s3d3828q6k2civ4ab1gfc7n-libXau-1.0.8
/nix/store/g8ymh2v9q1lc2zdzsj0986g5jzbv6i4a-libXdmcp-1.1.2
/nix/store/pb3rxnmp77wb7h7wijp5fyar4zrpl8ip-libxcb-1.11.1
/nix/store/if95xly2nygkvgsmb3xbn17d2p1kzh34-libX11-1.6.3
/nix/store/95y8cxz32vy4077dzd5jggg8l3x6zs8c-libXext-1.3.3
/nix/store/4y3w7mkl2a03gdhkarwss24r6m75mhbd-libXinerama-1.1.3
/nix/store/fpnf809d1dzq4dy0m7clbgn97fm3xxyd-bash-4.3-p42
/nix/store/7dq8cbpbagmn4v38sbvqmfl469rmqwcw-libXrender-0.9.9
/nix/store/xcvj60a669vm2mzsh29d7kim1fhsidal-zlib-1.2.8
/nix/store/ajva20yvb3wdkwq2ghv3hmfy20ql4xjm-libpng-apng-1.6.20
/nix/store/cl0ajhx1d5j5lg9q6zfh521cn1xq9lfm-bzip2-1.0.6
/nix/store/97bpv541iyydvp0qc1vnvdka3grzwspb-freetype-2.6.2
/nix/store/8qk3p1wrh3965sai3andxmpayibpp933-font-bh-ttf-1.0.3
/nix/store/j9bszda3d89zlxx9qdgpn2b44z3ywa3f-expat-2.1.0
/nix/store/i801jlsjxfg48kq0dlvh9fb30jpaiwpd-fontconfig-2.11.1-lib
/nix/store/h0lq0lc015nimn65wzv39y7k676pz4p0-libXft-2.3.2
/nix/store/24zsd7p4vpw3fickdzj33nwcqpb8fgbk-dmenu-4.6
/nix/store/f69rq0mmyp5mycgr6h8x0l252vgmq5vd-gcc-5.3.0-lib
/nix/store/j9ivh6h44rmlna2a28j94sxr71p6v072-j4-dmenu-desktop-2.14